src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -874,8 +875,10 @@ private double calculatePointsForTestCase(final ProgrammingExerciseTestCase test
         final double testPoints;
         double exerciseMaxPoints = scoreCalculationData.exercise().getMaxPoints();
 
-        // A weight-sum of zero would let the solution show an error to the instructor as the solution score must be
-        // 100% of all reachable points. To prevent this, we weigh all test cases equally in such a case.
+        // In case of a weight-sum of zero the instructor must be able to distinguish between a working solution
+        // (all tests passed, 0 points) and a solution with test failures.
+        // Only the second case should show a warning while the first case is considered as 100%.
+        // Therefore, all test cases have equal weight in such a case.
```

Comment:
Should this really be equivalent to 100%? Instructors might be confused why they got 100% and if students get 100% as well.
We could also improve the warning in the client to check if there is at least one test that was not executed or that was not successful

Reply:
I'm also not completely happy with that solution, but that's out of the scope of this PR. We should come up with a solution that:
- Does not need to handle the solution results in a special way
- Does not show unnecessary warnings if the configuration is correct but the solution has a score < 100%. In this context, we could also check for cases where hidden tests were not executed (by Ares) and therefore lead to a score of < 100%, but all public tests pass.
- Allows the instructor to easily detect issues in the test cases or configuration.